src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpMakeMethodAsynchronousService.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.MakeMethodAsynchronous;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeMethodAsynchronous
+{
+    [ExportLanguageService(typeof(IMakeMethodAsynchronousService), LanguageNames.CSharp), Shared]
+    internal sealed class CSharpMakeMethodAsynchronousService : AbstractMakeMethodAsynchronousService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpMakeMethodAsynchronousService()
+        {
+        }
+
+        public override bool IsAsyncReturnType(ITypeSymbol type, KnownTaskTypes knownTaskTypes)
```

Comment:
Is this necessary? I don't see anything language specific here (unless you are planning to move the logic of adding `async` into the service).

Reply:
@genlu The existing implementation in `main` branch for `IsAsyncReturnType` differs in VB from C#.

https://github.com/dotnet/roslyn/blob/a02ee4eaf806dd596e929a664405b0daadd84830/src/Features/CSharp/Portable/MakeMethodAsynchronous/CSharpMakeMethodAsynchronousCodeFixProvider.cs#L48-L52

https://github.com/dotnet/roslyn/blob/a02ee4eaf806dd596e929a664405b0daadd84830/src/Features/VisualBasic/Portable/MakeMethodAsynchronous/VisualBasicMakeMethodAsynchronousCodeFixProvider.vb#L51-L53